### DISCUSSION TITLE
How to show AWS ECS task logs in the Dagster UI?

### DISCUSSION THREAD

**Discussion by ammerzon** [dagster-io/dagster#18363] [Q&A] [answered]
Hello everyone!

I'm currently working on integrating AWS ECS tasks with Dagster, and I'm facing a challenge in displaying the logs from these ECS tasks within the Dagster UI. I'd appreciate any insights or guidance on how to achieve this.

![CleanShot 2023-11-29 at 15 53 04@2x](https://github.com/dagster-io/dagster/assets/7319168/b980cc82-d287-4660-b1e5-7fba8dd14736)

Here's some context:

* I'm running Dagster pipelines that trigger tasks in AWS ECS by following this [guide](https://docs.dagster.io/deployment/guides/aws)
* I want to be able to view the logs generated by these ECS tasks directly in the Dagster UI for easier monitoring and debugging.

My questions are:

* Is there a built-in feature in Dagster to support this kind of log integration?
* If not, what would be the best approach to implement this? Are there any recommended practices or tools that could help?

**Comment by gibsondan** [comment, marked as answer]
Hi @ammerzon - you can accomplish this by setting up a compute log manager and adding it to your dagster.yaml file: https://docs.dagster.io/deployment/dagster-instance#compute-log-storage - the s3 compute log manager is probably your best bet if you're on AWS.

**Reply by ammerzon** [reply]
Thank you. I wasn't aware that there is already an implementation for that 😁

**Reply by ammerzon** [reply]
@gibsondan what is the difference between using the [`cloudwatch`](https://github.com/dagster-io/dagster/blob/ae14b9e88207f1d22f7eff961eefb4e74a1cae84/python_modules/libraries/dagster-aws/dagster_aws/cloudwatch/loggers.py#L194) logger then? Are both shown in the UI?

**Reply by ammerzon** [reply]
One important note. I also had to give the Dagit `webserver` component access rights to the S3 bucket.
